### PR TITLE
03 image classification

### PR DESCRIPTION
**Pull request by jacquesqiao**
project: https://github.com/PaddlePaddle/Paddle/issues/4531
issue: https://github.com/PaddlePaddle/Paddle/issues/5100

- [x] batch_norm_layer
- [x] reshape_layer
- [x] img_conv_group
- [x] vgg16_bn_drop network
- [x] train_file
- [x] CPU && GPU train
- [x] loss reduce
- [ ] resnet network

### CODE REVIEW

File: paddle/framework/tensor.h [hunk 1]
```diff
@@ -130,7 +130,10 @@ class Tensor {
     return holder_->place();
   }
 
-  std::type_index type() const { return holder_->type(); }
+  std::type_index type() const {
+    PADDLE_ENFORCE_NOT_NULL(holder_, "Tensor get type() must contains holder");
```
Comment (wangkuiyi) [inline paddle/framework/tensor.h]: > "Tensor get type() must contains holder"

==>

"Tensor not initialized yet when Tensor::type() is called."

File: paddle/operators/batch_norm_op.cc [hunk 2]
```diff
@@ -64,6 +65,9 @@ class BatchNormOp : public framework::OperatorWithKernel {
         (tensor_format == TensorFormat::NCHW ? x_dims[1]
                                              : x_dims[x_dims.size() - 1]);
 
+    PADDLE_ENFORCE(x_dims.size() >= 3 && x_dims.size() <= 5,
+                   "input x dims should be between 3 and 5, actual");
```
Comment (wangkuiyi) [inline paddle/operators/batch_norm_op.cc]: Input x must have 3 to 5 dimensions.

File: python/paddle/v2/framework/tests/test_recognize_digits_conv.py [hunk 3]
```diff
@@ -86,6 +86,7 @@
                        fetch_list=[avg_cost])
 
         loss = np.array(outs[0])
+        print(loss)
```
Comment (wangkuiyi) [inline python/paddle/v2/framework/tests/test_recognize_digits_conv.py]: Should we remove or comment out this line?
